configure

```diff
@@ -0,0 +1,539 @@
+                                    output_location)))
+            success = False
+    return (success, messages)
+
+
+def generate_configs(mapping, dry_run, backup=True):
+    """
+     ... give some background on `mapping` and what it is doing
+     ...
+    """
+    print 'Testing for preconditions...\n'
+    success, messages = _handle_preconditions(mapping)
+    print_messages(messages)
+    if not success:
+        return (success, 1)
+    else:
+      print '%s%sPASSED%s\n\n' % (Fore.BLACK, Back.GREEN, Style.RESET_ALL)
+
+    print 'Generating configs...\n'
+    success, messages = _generate_configs(mapping, dry_run, backup)
+    print_messages(messages)
+    if not success:
+        return (success, 2)
+
+    return (success, 0)
+
+def _determine_message(t):
+    level = ''
+    if t == 'i':
+        level = Fore.WHITE
+    elif t == 'e':
+        level = Fore.RED
+    elif t == 'w':
+        level = Fore.YELLOW
+    return level
+
+def print_messages(messages):
+    for t, m in messages:
+        level = _determine_message(t)
+        print '%s%s%s' % (level, m, Style.RESET_ALL)
+    print '\n'
+
+
+def print_configs(mapping):
+    """
+    Print configuration file's name, template and output file information.
+    """
+    print 'Config Variables from %s' % (VARIABLES_PATH)
+    variables, messages = _get_variables()
+    if not variables:
+        print_messages(messages)
+    else:
+        pprint(variables)
+    print
+    print "Sections: %s\n" % (', '.join(mapping[0]))
+    print 'Config Name:\n\tTemplate => Output'
+    configs, messages = _get_filtered_config_files(mapping[1])
+    if not configs:
+        print_messages(messages)
+
+    for name in mapping[1]:
+        template, output = CONFIG_FILES[name]
+        print name + ':'
+        print '\t %s => %s' % (template, output)
+    print completed()
+
+
+def print_test(mapping):
+    """
+    Print results of testing configs preconditions. If there are no
+    problems acknowledge otherwise handle_precondition will print
+    the error(s).
+    """
+    print 'Testing for preconditions...\n'
+
+    c_files, messages = _get_filtered_config_files(mapping)
+    if not c_files:
+        print_messages(messages)
+        return
+    for section in mapping[0]:
+        print "section: %s" % (section)
+
+        for name in SECTION_MAPPING[section]:
+
+            m = ([section], [name])
+            success, messages = _handle_preconditions(m)
+            if success:
+                print '\t%s (file: %s) looks good.' % (name, CONFIG_FILES[name][1])
+            else:
+                print_messages(messages)
+        print
+    print completed()
+
+
+def print_verify():
+    """
+    Prints any variables present in the distribution files that are *not*
+    present in `VARIABLES_FILENAME`.
+    """
+    print 'Comparing %s with distribution file...\n' % (VARIABLES_FILENAME)
+    success, messages = _verify_variables_file()
```

Comment:
This message can get 'lost in the logs' when it happens *before* the output of generating distribution files, it might be better to hold the 'warning style' messages until the end for better visibility.